Tasks/Common/sdkutils.ts

```diff
@@ -26,32 +27,29 @@ export abstract class SdkUtils {
     public static readResources(): void {
         const taskManifestFile = path.join(__dirname, 'task.json')
         tl.setResourcePath(taskManifestFile)
-        SdkUtils.setSdkUserAgentFromManifest(taskManifestFile)
     }
 
     public static readResourcesFromRelativePath(relativeResourcePath: string): void {
         const taskManifestFile = path.join(__dirname, relativeResourcePath)
         tl.setResourcePath(taskManifestFile)
-        SdkUtils.setSdkUserAgentFromManifest(taskManifestFile)
     }
 
     // Injects a custom user agent conveying extension version and task being run into the
     // sdk so usage metrics can be tied to the tools.
-    public static setSdkUserAgentFromManifest(taskManifestFilePath: string): void {
-        if (fs.existsSync(taskManifestFilePath)) {
-            const agentVersion = process.env.AGENT_VERSION
-            const taskManifest = JSON.parse(fs.readFileSync(taskManifestFilePath, 'utf8')) as VSTSTaskManifest
-            const version = taskManifest.version
-            const userAgentString = `${this.userAgentPrefix}/${version.Major}.${version.Minor}.${version.Patch} ${
-                    this.userAgentSuffix
-                }${agentVersion}-${taskManifest.name}`
-                // tslint:disable-next-line:whitespace align
-            ;((AWS as any).util as any).userAgent = () => {
-                return userAgentString
-            }
-        } else {
+    public static getCustomUserAgentFromManifest(taskManifestFilePath: string): string {
+        if (!fs.existsSync(taskManifestFilePath)) {
             console.warn(`Task manifest ${taskManifestFilePath} not found, cannot set custom user agent!`)
+
+            return undefined
         }
+        const agentVersion = process.env.AGENT_VERSION
+        const taskManifest = JSON.parse(fs.readFileSync(taskManifestFilePath, 'utf8')) as VSTSTaskManifest
+        const version = taskManifest.version
+        const userAgentString = `${this.userAgentPrefix}/${version.Major}.${version.Minor}.${version.Patch} ${
+            this.userAgentSuffix
+        }${agentVersion}-${taskManifest.name}`
```

Comment:
PR #235 replaces the dash with a space.

Reply:
Yes it does, these do two different things and need to be merged

Reply:
It also completely changes the format  :P